tracee-ebpf/external/external_test.go

```diff
@@ -103,3 +104,118 @@ func TestArgumentUnmarshalJSON(t *testing.T) {
 		}
 	}
 }
+
+func TestEvent_ToUnstructured(t *testing.T) {
+	testCases := []struct {
+		name  string
+		event Event
+	}{
+		{
+			name:  "Should unstructure zero Event",
+			event: Event{},
+		},
+		{
+			name: "Should unstructure Event with empty slices",
+			event: Event{
+				Args:           []Argument{},
+				StackAddresses: []uint64{},
+			},
+		},
+		{
+			name: "should unstructure args",
+			event: Event{
+				Args: []Argument{
+					{
+						ArgMeta: ArgMeta{
+							Name: "dirfd",
+							Type: "int",
+						},
+						Value: -100,
+					},
+					{
+						ArgMeta: ArgMeta{
+							Name: "pathname",
+							Type: "const char",
+						},
+						Value: "/sys/fs/cgroup/cpu,cpuacct/cpuacct.stat",
+					},
+					{
+						ArgMeta: ArgMeta{
+							Name: "flags",
+							Type: "int",
+						},
+						Value: "O_RDONLY|O_CLOEXEC",
+					},
+					{
+						ArgMeta: ArgMeta{
+							Name: "mode",
+							Type: "mode_t",
+						},
+						Value: 5038682,
+					},
+				},
+			},
+		},
+		{
+			name: "Should unstructure Event",
+			event: Event{
+				Timestamp:           7126.141189,
+				ProcessID:           1,
+				ThreadID:            1,
+				ParentProcessID:     4798,
+				HostProcessID:       4819,
+				HostThreadID:        4819,
+				HostParentProcessID: 4798,
+				UserID:              0,
+				MountNS:             4026532256,
+				PIDNS:               4026532259,
+				ProcessName:         "cadvisor",
+				HostName:            "4213291591ab",
+				EventID:             257,
+				EventName:           "openat",
+				ArgsNum:             4,
+				ReturnValue:         14,
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			actual, err := tc.event.ToUnstructured()
+			requireNoError(t, err)
+			expected, err := jsonRoundTrip(tc.event)
+			requireNoError(t, err)
+
+			if !reflect.DeepEqual(expected, actual) {
+				t.Errorf("Not equal:\nwant: %v\nhave: %v\n", expected, actual)
+			}
```

Comment:
Could we use the stretchr/testify assert.Equals instead? It's a little easier to read in the output. I believe at the moment the project has a mix of both but IMO we should use stretchr/testify for readability.

Reply:
Done